custom/icds_reports/utils/aggregation_helpers/distributed/agg_service_delivery_report.py

```diff
@@ -0,0 +1,256 @@
+            ('aggregation_level', '5'),
+            ('state_is_test','awc_location_local.state_is_test'),
+            ('district_is_test','awc_location_local.district_is_test'),
+            ('block_is_test','awc_location_local.block_is_test'),
+            ('supervisor_is_test','awc_location_local.supervisor_is_test'),
+            ('awc_is_test','awc_location_local.awc_is_test'),
+
+        )
+        yield """
+                INSERT INTO "{tmp_tablename}" (
+                    {columns}
+                )
+                (
+                SELECT
+                {calculations}
+                from awc_location_local 
+                WHERE awc_location_local.aggregation_level=5 AND awc_location_local.state_is_test<>1);
+                """.format(
+            tmp_tablename=self.staging_tablename,
+            columns=", ".join([col[0] for col in columns]),
+            calculations=", ".join([col[1] for col in columns])
+
+        ), {
+            'start_date': self.month
+        }
+
+
+        yield """
+        CREATE TABLE temp_child_table AS
+        SELECT
+            supervisor_id,
+            awc_id,
+            SUM(pse_eligible) as pse_eligible,
+            SUM(CASE WHEN pse_eligible=1 AND pse_days_attended=0 THEN 1 ELSE 0 END) as pse_0_days,
+            SUM(CASE WHEN pse_eligible=1 AND pse_days_attended BETWEEN 1 AND 7 THEN 1 ELSE 0 END) as pse_1_7_days,
+            SUM(CASE WHEN pse_eligible=1 AND pse_days_attended BETWEEN 8 AND 14 THEN 1 ELSE 0 END) as pse_8_14_days,
+            SUM(CASE WHEN pse_eligible=1 AND pse_days_attended BETWEEN 15 AND 20 THEN 1 ELSE 0 END) as pse_15_20_days,
+            SUM(CASE WHEN pse_eligible=1 AND pse_days_attended>=21 THEN 1 ELSE 0 END) as pse_21_days,
+            
+            SUM(CASE WHEN pse_eligible=1 AND lunch_count=0 THEN 1 ELSE 0 END) as lunch_0_days,
+            SUM(CASE WHEN pse_eligible=1 AND lunch_count BETWEEN 1 AND 7 THEN 1 ELSE 0 END) as lunch_1_7_days,
+            SUM(CASE WHEN pse_eligible=1 AND lunch_count BETWEEN 8 AND 14 THEN 1 ELSE 0 END) as lunch_8_14_days,
+            SUM(CASE WHEN pse_eligible=1 AND lunch_count BETWEEN 15 AND 20 THEN 1 ELSE 0 END) as lunch_15_20_days,
+            SUM(CASE WHEN pse_eligible=1 AND lunch_count>=21 THEN 1 ELSE 0 END) as lunch_21_days,
+            
+            SUM(thr_eligible) as thr_eligible,
+            SUM(CASE WHEN thr_eligible=1 AND num_rations_distributed=0 THEN 1 ELSE 0 END) as thr_0_days,
+            SUM(CASE WHEN thr_eligible=1 AND num_rations_distributed BETWEEN 1 AND 7 THEN 1 ELSE 0 END) as thr_1_7_days,
+            SUM(CASE WHEN thr_eligible=1 AND num_rations_distributed BETWEEN 8 AND 14 THEN 1 ELSE 0 END) as thr_8_14_days,
+            SUM(CASE WHEN thr_eligible=1 AND num_rations_distributed BETWEEN 15 AND 20 THEN 1 ELSE 0 END) as thr_15_20_days,
+            SUM(CASE WHEN thr_eligible=1 AND num_rations_distributed>=21 THEN 1 ELSE 0 END) as thr_21_days
+        FROM child_health_monthly
```

Comment:
this should pull from the temp child health table (the same one agg_awc and agg_child_health pull from)